### ISSUE TITLE
Add simple string processing option

### ISSUE THREAD

**Issue by yagarea** [NiklasEi/jekyll_custom_permalink#1]
Hello, it would be very nice if your plugin had option for:
- replacing white spaces with `-` 
- removing diacritics
- Setting character to lowercase

Example (:town = "Žďár nad Sázavou"):  
`/city/:town/` => `/city/zdar-nad-sazavou/`

**Comment by yagarea** [comment]
this method should do the trick:
```ruby
def self.process_string_for_url(raw_string)
    raw_string.tr("ÀÁÂÃÄÅàáâãäåĀāĂăĄąÇçĆćĈĉĊċČčÐðĎďĐđÈÉÊËèéêëĒēĔĕĖėĘęĚěĜĝĞğĠġĢģĤĥĦħÌÍÎÏìíîïĨĩĪīĬĭĮįİıĴĵĶķĸĹĺĻļĽľĿŀŁłÑñŃńŅņŇňŉŊŋÒÓÔÕÖØòóôõöøŌōŎŏŐőŔŕŖŗŘřŚśŜŝŞşŠšſŢţŤťŦŧÙÚÛÜùúûüŨũŪūŬŭŮůŰűŲųŴŵÝýÿŶŷŸŹźŻżŽž",
"AAAAAAaaaaaaAaAaAaCcCcCcCcCcDdDdDdEEEEeeeeEeEeEeEeEeGgGgGgGgHhHhIIIIiiiiIiIiIiIiIiJjKkkLlLlLlLlLlNnNnNnNnnNnOOOOOOooooooOoOoOoRrRrRrSsSsSsSssTtTtTtUUUUuuuuUuUuUuUuUuUuWwYyyYyYZzZzZz").gsub(/\s/, "-").downcase
end
```

**Comment by NiklasEi** [comment]
Couldn't we use something like https://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-transliterate?
I haven't touched ruby in years and don't have a test setup for this plugin anymore, but I would prefer a solution without giant lists of hardcoded characters.

Maybe https://api.rubyonrails.org/classes/ActiveSupport/Inflector.html#method-i-parameterize would be easier since it seems to also check your first and third point.

**Comment by yagarea** [comment]
I completely understand that you do not want to implement this abomination, but it is only thing I got working. I was trying to use parametrize but it looks like plain ruby does not have it and it is in framework ruby on rails. This method is only thing I got somewhat working.

I forked your plugin to make pull request but I never have never written anything in ruby in my life so spend one evening on it and than just made an issue. even I made the method above I was not able to put in to the plugin it self.  I am afraid I am not able to do it without your assistance.

**Comment by NiklasEi** [comment]
We might learn a bit or two from ruby on rails [parameterize](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/inflector/transliterate.rb#L121-L145) and [transliterate](https://github.com/rails/rails/blob/main/activesupport/lib/active_support/inflector/transliterate.rb#L64-L93) implementations. I will see if I can set up Ruby again so I can play around with this.

**Comment by NiklasEi** [comment]
Resolved in #2
